### PR TITLE
Gracefully handle logbook retrieval errors

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

Related to this issue: https://github.com/home-assistant/core/issues/49180. A separate PR for the root cause is coming shortly.

If for some reason the logbook entries cannot be retrieved, gracefully handle that situation instead of showing endless loading spinners.

**More-info:**

image

**Logbook panel (I opted for a alert popup to clearly inform the user, whereas in the more-info I feel that is too invasive):**
Note: The actual error message will be shown 1:1 as coming from the backend, so potentially not localized.

image

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/cards/hui-logbook-card.ts [hunk 1]
```diff
@@ -229,23 +237,27 @@ export class HuiLogbookCard extends LitElement implements LovelaceCard {
...
-    const logbookEntries = this._logbookEntries
-      ? [...newEntries, ...this._logbookEntries]
-      : newEntries;
-
-    this._logbookEntries = logbookEntries.filter(
-      (logEntry) => new Date(logEntry.when) > hoursToShowDate
-    );
-
-    this._lastLogbookDate = now;
+    try {
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-logbook-card.ts:240]: Only put the try catch around the `getLogbookData` call

Comment (spacegaier) [reply src/panels/lovelace/cards/hui-logbook-card.ts:244]: I have narrowed now the try/catch scopes or did you mean to split up the `Promise.all()` calls into separate ones so we can really only try/catch `getLogbookDate()`? Then we would loose the fail-fast behavior.

File: src/dialogs/more-info/ha-more-info-logbook.ts [hunk 2]
```diff
@@ -29,6 +29,8 @@ export class MoreInfoLogbook extends LitElement {
 
   private _fetchUserPromise?: Promise<void>;
 
+  private _error = false;
```
Comment (bramkragten) [inline src/dialogs/more-info/ha-more-info-logbook.ts:32]: We normally make `_error` a string, so we can handle multiple errors.

File: src/dialogs/more-info/ha-more-info-logbook.ts [hunk 3]
```diff
@@ -119,17 +125,25 @@ export class MoreInfoLogbook extends LitElement {
...
+          lastDate.toISOString(),
+          now.toISOString(),
+          this.entityId,
+          true
+        ),
+        loadTraceContexts(this.hass),
+        this._fetchUserPromise,
+      ]);
+    } catch (err) {
+      this._error = true;
```
Comment (bramkragten) [inline src/dialogs/more-info/ha-more-info-logbook.ts:144]:
```suggestion
      this._error = err.message;
```

File: src/panels/lovelace/cards/hui-logbook-card.ts [hunk 4]
```diff
@@ -231,17 +239,22 @@ export class HuiLogbookCard extends LitElement implements LovelaceCard {
...
+          this.hass,
+          lastDate.toISOString(),
+          now.toISOString(),
+          this._configEntities!.map((entity) => entity.entity).toString(),
+          true
+        ),
+        this._fetchUserPromise,
+      ]);
+    } catch (err) {
+      this._error = true;
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-logbook-card.ts:256]:
```suggestion
      this._error = err.message;
```

File: src/panels/lovelace/cards/hui-logbook-card.ts [hunk 5]
```diff
@@ -187,7 +189,13 @@ export class HuiLogbookCard extends LitElement implements LovelaceCard {
         class=${classMap({ "no-header": !this._config!.title })}
       >
         <div class="content">
-          ${!this._logbookEntries
+          ${this._error
+            ? html`
+                <div class="no-entries">
+                  ${this.hass.localize("ui.components.logbook.retrieval_error")}
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-logbook-card.ts:195]:
```suggestion
                  ${`${this.hass.localize("ui.components.logbook.retrieval_error")} ${this._error}`}
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**